go/enclave/l2chain/l2_chain.go

```diff
@@ -818,22 +817,40 @@ func (oc *ObscuroChain) processRollups(block *common.L1Block) error {
 		if err = oc.storage.StoreRollup(rollup); err != nil {
 			return fmt.Errorf("could not store rollup. Cause: %w", err)
 		}
-	}
 
-	newHeadRollup := currentHeadRollup
-	if len(rollups) > 0 {
-		newHeadRollup = rollups[len(rollups)-1]
-	}
-	if newHeadRollup != nil {
-		l1Head := block.Hash()
-		if err = oc.storage.UpdateHeadRollup(&l1Head, newHeadRollup.Hash()); err != nil {
-			return fmt.Errorf("could not update L2 head rollup. Cause: %w", err)
+		blockHash := block.Hash()
+		if err = oc.storage.UpdateHeadRollup(&blockHash, rollup.Hash()); err != nil {
+			return fmt.Errorf("could not update head rollup. Cause: %w", err)
 		}
 	}
 
 	return nil
 }
 
+// Given a block, returns the latest rollup in the canonical chain for that block (excluding those in the block itself).
```

Comment:
Maybe rename the method to getLatestRollupBefore(block) or something. I was briefly confused as to why it was excluding current block.